lambdas/es/indexer/test/test_index.py

```diff
@@ -19,130 +22,216 @@
+        *,
+        bucket="test-bucket",
+        eTag="123456",
+        key="hello+world.txt",
+        region="us-east-1",
+        size=100,
+        versionId="1313131313131.Vier50HdNbi7ZirO65",
+        bucket_versioning=True
+):
+    """return an event based on EVENT_CORE, add fields to match organic AWS events"""
+    if name in {
+            "ObjectCreated:Put",
+            "ObjectCreated:Copy",
+            "ObjectCreated:Post",
+            "ObjectCreated:CompleteMultipartUpload"
+    }:
+        args = {}
+        args["bucket"] = bucket
+        args["eTag"] = eTag
+        args["key"] = key
+        args["region"] = region
+        args["size"] = size
+        if bucket_versioning:
+            args["versionId"] = versionId
+        return _make_event(
+            name,
+            **args
+        )
+    # no versionId or eTag in this case
+    elif name == "ObjectRemoved:Delete":
+        return _make_event(
+            name,
+            bucket=bucket,
+            key=key,
+            region=region
+        )
+    elif name == "ObjectRemoved:DeleteMarkerCreated":
+        # these events are possible in both versioned and unversioned buckets
+        # (e.g. bucket now unversioned that was versioned will generate a
+        # delete marker on `aws s3 rm`)
+        args = {}
+        args["bucket"] = bucket
+        args["eTag"] = eTag
+        args["key"] = key
+        args["region"] = region
+        args["size"] = size
+        if bucket_versioning:
+            args["versionId"] = versionId
+        return _make_event(
+            name,
+            **args
+        )
+    elif name == UNKNOWN_EVENT_TYPE:
+        return _make_event(UNKNOWN_EVENT_TYPE)
+
+    else:
+        raise ValueError(f"Unexpected event type: {name}")
+
+
+def _make_event(
+        name,
+        *,
+        bucket="",
+        eTag="",
+        key="",
+        region="",
+        size=0,
+        versionId=""
+):
+    """make events in the pattern of
+    https://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html
+    and
+    AWS Lambda > Console > Test Event
+    """
+    e = deepcopy(EVENT_CORE)
+    e["eventName"] = name
+
+    if bucket:
+        e["s3"]["bucket"]["name"] = bucket
+        e["s3"]["bucket"]["arn"] = f"arn:aws:s3:::{bucket}"
+    if key:
+        e["s3"]["object"]["key"] = key
+    if eTag:
+        e["s3"]["object"]["eTag"] = eTag
+    if size:
+        e["s3"]["object"]["size"] = size
+    if region:
+        e["awsRegion"] = region
+    if versionId:
+        e["s3"]["object"]["versionId"] = versionId
+
+    return e
+
+
 class MockContext():
     def get_remaining_time_in_millis(self):
         return 30000
 
 
 class TestIndex(TestCase):
     def setUp(self):
+        # TODO: figure out why this is required. If =True it will cause
+        # test_create_index_events to fail for no obvious reason
         self.requests_mock = responses.RequestsMock(assert_all_requests_are_fired=False)
```

Comment:
@sir-sigurd Can you tell why turning this off causes the failing mock to fail? AFAICT the mocked endpoint is called as many times as it is mocked.